src/main/webapp/3Dscript.js

```diff
@@ -0,0 +1,183 @@
+import {FlyControls} from 'https://threejs.org/examples/jsm/controls/FlyControls.js';
+import {OrbitControls} from 'https://threejs.org/examples/jsm/controls/OrbitControls.js';
+import {GUI} from 'https://threejs.org/examples/jsm/libs/dat.gui.module.js';
+
+// These are global objects.
+let scene;
+let camera;
+let renderer;
+let clock;
+let controls;
+let orientation;
+let trajectory;
+let dataLength;
+let pose;
+let poseRotation = {degrees: 0};
+let poseScalar = {multiplier: 1};
+let posePosition = {x: 0, z: 0};
+const apiKey = 'AIzaSyDCgKca9sLuoQ9xQDfHUvZf1_KAv06SoTU';
+
+initThreeJs();
+gui();
+animate();
+
+/**
+ * Initializes the essential three.js 3D components then fetchs 
+ * the pose data.
+ */
+function initThreeJs() {
+  scene = new THREE.Scene();
+  camera = new THREE.PerspectiveCamera(
+      /* fov =*/ 75,
+      /* aspectRatio =*/ window.innerWidth / window.innerHeight,
+      /* nearFrustum =*/.1,
+      /* farFrustum =*/ 1000);
+  camera.position.set(0, 1, 1);
+  renderer = new THREE.WebGLRenderer();
+  renderer.setSize(window.innerWidth, window.innerHeight);
+  document.body.appendChild(renderer.domElement);
+  clock = new THREE.Clock();
+  controls = new OrbitControls(camera, renderer.domElement );
+
+  fetchData();
+}
+
+
+/**
+ * This function fetchs pose data from the RunInfo servlet,
+ * it is an asynchronous call requiring addTerrain() to be 
+ * called after the data is fully loaded.
+ */
+function fetchData() {
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    const id = urlParams.get('id');
+    const type = urlParams.get('dataType');
+    fetch('/getrun?id=' + id + '&dataType=' + type)
+    .then(response => response.json())
+    .then(data => pose = data)
+    .then(() => addTerrian());
+}
+
+/**
+ * Creates the 2D terrain and points a light at it.
+ * It also calls two other functions so it can wait on the fetch 
+ * request and significantly cut down rendering time with 
+ * geometry instancing.
+*/
+function addTerrian() {
```

Comment:
We typically use "terrain" to indicate that the surface includes altitude data. Maybe just call this "map" since it's 2D?